personalities/sctfe/config.go

```diff
@@ -65,12 +122,39 @@ func LogConfigFromFile(filename string) (*configpb.LogConfig, error) {
 //   - Merge delays (if present) are correct.
 //
 // Returns the validated structures (useful to avoid double validation).
-func ValidateLogConfig(cfg *configpb.LogConfig) (*ValidatedLogConfig, error) {
-	if len(cfg.Origin) == 0 {
-		return nil, errors.New("empty log origin")
+func ValidateLogConfig(cfg *configpb.LogConfig, origin string, projectID string, bucket string, spannerDB string) (*ValidatedLogConfig, error) {
+	if len(origin) == 0 {
+		return nil, errors.New("empty origin")
+	}
+
+	// TODO(phboneff): move this logic together with the tests out of config.go and validate the flags directly
+	if len(projectID) == 0 {
+		return nil, errors.New("empty projectID")
+	}
+
+	if len(bucket) == 0 {
+		return nil, errors.New("empty bucket")
+	}
+
+	if len(spannerDB) == 0 {
+		return nil, errors.New("empty spannerDB")
 	}
```

Comment:
Yeah, these ones should really be in the `main.go` since it's GCP specific. (just agreeing with the comment, really :) )